mdoc/src/main/scala-2/mdoc/internal/markdown/Instrumenter.scala

```diff
@@ -175,8 +175,8 @@ object Instrumenter {
     val wrapped = new StringBuilder()
       .append("package repl\n")
       .append("object MdocSession extends _root_.mdoc.internal.document.DocumentBuilder {\n")
-      .append("  def app(): _root_.scala.Unit = {val _ = new MdocApp()}\n")
-      .append("  class MdocApp {\n")
+      .append("  def app(): _root_.scala.Unit = {val _ = MdocApp}\n")
+      .append("  object MdocApp {\n")
```

Comment:
So I think this indeed solves the current issue, but I wonder if we went full circle here. There were similar issue within Scala CLI, we could at least see if those cases work. The related PRs in scala CLI https://github.com/VirtusLab/scala-cli/issues?q=is%3Aissue+script+is%3Aclosed+label%3Ascript-wrapper

Potentially we could use scalameta to put any classes/object outside the MdocApp class as an alternative, but that is might be a lot of additional work

Reply:
I figured there is probably a reason it is a class not an object but I hoped one of the tests would indicate why. I'll add such a test.

> Potentially we could use scalameta to put any classes/object outside the MdocApp class as an alternative, but that is might be a lot of additional work

I shouldn't be that much work... I'll try to do it, I'd be nice to be done with this whole family of problems.

Reply:
Would be useful to share it somehow between Scala CLI and here. We can add it as a separate library.